app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -1,50 +1,123 @@
     private lateinit var binding: UserProfileFragmentBinding
+    private var currentUserId: String? = ""
+    private lateinit var imageAccess: ImageAccess
+    private var imageUri: Uri? = null
     @Inject
     lateinit var currentUserProvider: CurrentUserProvider
+    @Inject
+    lateinit var userRepository: UserRepository
+    private val TESTURI = "https://picsum.photos/200"
+
+    companion object {
+        fun newInstance() = UserProfileFragment()
+        var isTestCase: Boolean = false
+    }
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         binding = UserProfileFragmentBinding.inflate(inflater, container, false)
-
+        currentUserId = currentUserProvider.getCurrentUserId()
         // If no userId is provided, we get the user that is currently logged in.
         val userId = when(args.userId) {
-                null, "" -> currentUserProvider.getCurrentUserId()
+            null, "" -> currentUserId
                 else -> args.userId
-        }
 
+        }
+        isTestCase = false
+        imageAccess = ImageAccess(requireActivity())
+        imageAccess.setupImageView(binding.imageView)
+        lifecycle.addObserver(imageAccess)
         viewModel.setUser(userId)
-
         viewModel.user.observe(viewLifecycleOwner, { user ->
             if (user != null) {
                 binding.nameText.text = user.name
                 Glide.with(this).load(user.profilePicture).into(binding.imageView)
             }
         })
-
         binding.viewModel = viewModel
-
+        setupButtonsVisibility()
         return binding.root
     }
 
+    private fun setupButtonsVisibility() {
+        currentUserId = currentUserProvider.getCurrentUserId()
+        val buttons = listOf(binding.btnOpenGallery, binding.btnOpenCamera, binding.btnApply)
+        for(button: Button in buttons) {
+            button.visibility =
+                if(currentUserId != null) View.VISIBLE
+                else View.GONE
+        }
+        binding.btnApply.visibility = View.GONE
+        setupButtons()
+
+    }
+    private fun setupButtons() {
+        val buttons = listOf(binding.btnApply, binding.btnOpenCamera, binding.btnOpenGallery)
+        for(button: Button in buttons) {
+            button.setOnClickListener {
+                binding.btnApply.visibility =
+                    if(button == binding.btnOpenCamera || button == binding.btnOpenGallery)
+                        View.VISIBLE
+                    else View.GONE
+                getAction(button)
+            }
+        }
+    }
+
+    private fun getAction(button: Button) {
+        when (button) {
+            binding.btnOpenCamera -> imageAccess.openCamera()
+            binding.btnOpenGallery -> imageAccess.openGallery()
+            binding.btnApply -> {
+                imageUri = if (!isTestCase) imageAccess.getImageUri() else Uri.parse(TESTURI)
```

Comment:
As mentioned in the test, I am not a huge fan of this testCase variable. Isn't there a way to avoid that?